omnipaxos_storage/src/persistent_storage.rs

```diff
@@ -102,76 +110,132 @@ impl PersistentStorageConfig {
+            rocksdb_options: {
+                let mut opts = Options::default();
+                opts.create_if_missing(true);
+                opts
+            },
+            #[cfg(not(feature = "rocksdb"))]
+            sled_options: Config::new(),
         }
     }
 }
 
 /// A persistent storage implementation, lets sequence paxos write the log
-/// and variables (e.g. promised, accepted) to disk. Log entries are serialized
-/// and de-serialized into slice of bytes when read or written from the log.
+/// and current state to disk. Log entries are serialized and de-serialized
+/// into slice of bytes when read or written from the log.
 pub struct PersistentStorage<T, S>
 where
     T: Entry,
     S: Snapshot<T>,
 {
-    /// disk-based commit log for entries
+    /// Disk-based commit log for entries
     commitlog: CommitLog,
-    /// the path to the directory containing a commitlog
+    /// The path to the directory containing a commitlog
     log_path: String,
-    /// local RocksDB key-value store
+    /// Local RocksDB key-value store, must be enabled as a feature
+    #[cfg(feature = "rocksdb")]
     rocksdb: DB,
+    /// Local sled key-value store, enabled by default
+    #[cfg(not(feature = "rocksdb"))]
+    sled: Db,
     /// A placeholder for the T: Entry
     t: PhantomData<T>,
     /// A placeholder for the S: Snapshot<T>
     s: PhantomData<S>,
 }
 
 impl<T: Entry, S: Snapshot<T>> PersistentStorage<T, S> {
-    pub fn with(storage_config: PersistentStorageConfig) -> Self {
+    // Creates or opens an existing storage
+    pub fn open(storage_config: PersistentStorageConfig) -> Self {
         let path = storage_config.path.expect("No path found in config");
 
-        // Initialize Commitlog and rocksDB
         let commitlog =
             CommitLog::new(storage_config.commitlog_options).expect("Failed to create Commitlog");
-        let rocksdb = DB::open(&storage_config.rocksdb_options, format!("{path}{ROCKSDB}"))
-            .expect("Failed to create rocksDB store");
 
         Self {
             commitlog,
             log_path: format!("{path}{COMMITLOG}"),
-            rocksdb,
+            #[cfg(feature = "rocksdb")]
+            rocksdb: {
+                DB::open(&storage_config.rocksdb_options, format!("{path}{DATABASE}"))
+                    .expect("Failed to create rocksDB database")
+            },
+            #[cfg(not(feature = "rocksdb"))]
+            sled: {
+                let opts = Config::path(storage_config.sled_options, format!("{path}{DATABASE}"));
+                Config::open(&opts).expect("Failed to create sled database")
+            },
             t: PhantomData::default(),
             s: PhantomData::default(),
         }
     }
+
+    // Creates a new storage instance, panics if a commitlog or rocksDB/sled instance exists in the given path
+    pub fn new(storage_config: PersistentStorageConfig) -> Self {
+        let path = storage_config
+            .path
+            .clone()
```

Comment:
I don't think you need to clone here:
```suggestion
            .as_ref()
```